plugins/check_smtp.c

```diff
@@ -632,10 +649,14 @@ process_arguments (int argc, char **argv)
 #else
 			usage (_("SSL support not available - install OpenSSL and recompile"));
 #endif
-			// fall through
+		case 's':
+		/* ssl */
+			use_ssl = TRUE;
+			server_port = SMTPS_PORT;
```

Comment:
this will overwrite a previous `--port` option.

Reply:
Good catch, @RincewindsHat . Thanks! I added another commit to move the port option more to the bottom. Thus `-p` takes precedence over the default port with `-s`/`--ssl`.

Reply:
Thanks @franzs, but this does not change the behaviour, since the `getopt` loop executes for every option, the position in the `switch` `case` does not matter at all (except for fall througs).
Could you undo that last change and perhaps use some local variable (let's say `bool port_explicitely_set`) to determine whether the port was set explicitely via the option?

Reply:
:man_facepalming:
Yes, of course. It was too late yesterday... :innocent:
The order of command line options matters and not the order of the case blocks.
Thanks for pointing out, @RincewindsHat
I changed this commit. Please have a look at it.